src/docs/files-provider-deprecation.rst

```diff
@@ -0,0 +1,60 @@
+Files Provider Removal
+######################
+
+The SSSD team has announced the `removal of the files provider <https://pagure.io/fesco/issue/3107>`_ feature in recent versions of SSSD in Fedora/CentOS Stream/RHEL. This document describes how to switch to using the SSSD 'proxy' provider to replace files provider functionality.
+
+Practically, there are only two use cases that currently justify usage of the files provider feature:
+
+* smart card authentication of local users
+* session recording for local users
+
+For both cases the proxy provider is a viable substitute. Otherwise, files provider functionality does not provide any benefit and should remain disabled.
+
+Proxy Provider Configuration
+****************************
+
+The SSSD proxy provider is just a relay, an intermediary configuration. SSSD connects to its proxy service, and then that proxy loads the specified libraries. This allows SSSD to use some resources that it otherwise would not be able to use.
+
+First, make sure to install the ``sssd-proxy`` package.
+
+.. code-block:: console
+
+    $ dnf install sssd-proxy
+
+A sample proxy provider configuration to load the nss `files` module is below:
+
+.. code-block:: console
+
+    [sssd]
+    services=nss, pam
+    domains=nssfiles
+
+    [domain/nssfiles]
+    id_provider=proxy
+    proxy_lib_name=files
+    proxy_pam_target=sssd-shadowutils
+
+The ``proxy_lib_name`` option specifies which existing NSS library to proxy identity requests through.
+
+The ``proxy_pam_target`` specifies the target to which PAM must proxy as an authentication provider. This PAM target is a file containing PAM stack information in the default PAM directory, ``/etc/pam.d/``. The following ``sssd-shadowutils`` file exists on RHEL-based Linux systems.
+
+.. note:: Ensure that the proxy PAM target stack does not recursively include pam_sss.so.
+
+.. code-block:: console
+
+    ~# cat /etc/pam.d/sssd-shadowutils
+    #%PAM-1.0
+    auth        [success=done ignore=ignore default=die] pam_unix.so nullok try_first_pass
+    auth        required      pam_deny.so
+
+    account     required      pam_unix.so
+    account     required      pam_permit.so
+
+nsswitch changes
+****************
+
+In the `session recording for local users` case it is required set the `sss` module as the first module in the list for the `passwd` and `group` databases of ``/etc/nsswitch.conf``. The following command should be run:
+
+.. code-block:: bash
+
+    authselect select sssd with-tlog
```

Comment:
Maybe "could be used on a systems, managed by authselect"?

Current text, imo, is a bit too Fedora-centric for upstream resource...

Reply:
Fixed.